### DISCUSSION TITLE
Configuring a firewall with UFW

### DISCUSSION THREAD

**Discussion by sheshko-as** [Ylianst/MeshCentral#5827] [Q&A]
I have configured the firewall for my MeshCentral server, everything works correctly, there are no problems except: Check server version
If UFW is Enabled, then when you click Check, the server version freezes on this window:
![image](https://github.com/Ylianst/MeshCentral/assets/97667637/16091a4e-f862-4ec2-a16b-d82bfbf6a7a8)

UFW Settings:
ufw default deny incoming
ufw default allow outgoing
ufw allow https
ufw allow http
ufw allow proto tcp from XXX.XXX.XXX.XXX to any port 22
ufw enable && ufw reload

**Comment by si458** [comment]
im sure i replied to this before? but maybe i didnt?
the MeshCentral Version simply runs `npm dist-tag ls meshcentral` from the root of your meshcentral folder
(folder with node_modules, meshcentral-data, etc..)

so it might be your machine cant talk to npmjs.com OR they are having issues maybe?

**Reply by sheshko-as** [reply]
Please forgive me, maybe I didn't see it.
If you disable ufw with the command: ufw disable, then everything works correctly.

**Reply by si458** [reply]
this to me strikes something to do with ufw then?
but i dont use ufw as i use the hypervisor firewall (proxmox) and router firewalls instead so im not too sure?
you could check the logs and see if anything sticks out there when the firewall is on?
https://www.cherryservers.com/blog/how-to-configure-ubuntu-firewall-with-ufw#understand-ufw-logs

**Reply by sheshko-as** [reply]
The problem was that meshcentral was installed in \opt\meshcentral
